rosbag2_transport/src/rosbag2_transport/locked_priority_queue.hpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 Apex.AI, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_TRANSPORT__LOCKED_PRIORITY_QUEUE_HPP_
+#define ROSBAG2_TRANSPORT__LOCKED_PRIORITY_QUEUE_HPP_
+
+#include <mutex>
+#include <optional>
+#include <queue>
+
+#include "rcpputils/thread_safety_annotations.hpp"
+#include "rcpputils/unique_lock.hpp"
+
+/// \brief `std::priority_queue` wrapper with locks.
+/// \tparam T the element type
+/// \tparam Container the underlying container type
+/// \tparam Compare the comparator
+/// \see std::priority_queue
+template<typename T, typename Container, typename Compare>
```

Comment:
Suggestion to add a default values to the template parameters
```suggestion
template<typename T, typename Container = std::vector<T>,
  typename Compare = std::less<typename Container::value_type>>
```

Reply:
changed in ca23e7fae501e6a2950f529ef0dafa2f37dd1f26